code/datums/components/iff_fire_prevention.dm

```diff
@@ -0,0 +1,78 @@
+#define IFF_HALT_COOLDOWN 0.5 SECONDS
+
+/datum/component/iff_fire_prevention
+	var/iff_additional_fire_delay
+	COOLDOWN_DECLARE(iff_halt_cooldown)
+
+/datum/component/iff_fire_prevention/Initialize(additional_fire_delay = 0)
+	. = ..()
+	iff_additional_fire_delay = additional_fire_delay
+
+
+/datum/component/iff_fire_prevention/RegisterWithParent()
+	. = ..()
+	RegisterSignal(parent, COMSIG_GUN_BEFORE_FIRE, PROC_REF(check_firing_lane))
+
+/datum/component/iff_fire_prevention/UnregisterFromParent()
+	. = ..()
+	UnregisterSignal(parent, COMSIG_GUN_BEFORE_FIRE)
+
+/datum/component/iff_fire_prevention/proc/check_firing_lane(obj/firing_weapon, obj/projectile/projectile_to_fire, atom/target, mob/living/user)
+	SIGNAL_HANDLER
+
+	var/angle = get_angle(user, target)
+
+	var/range_to_check = user.get_maximum_view_range()
+
+	var/extended_target_turf = get_angle_target_turf(user, angle, range_to_check)
+
+
+	var/turf/starting_turf = get_turf(user)
+
+	if(!starting_turf || !extended_target_turf)
+		return COMPONENT_CANCEL_GUN_BEFORE_FIRE
+
+	var/list/checked_turfs = getline2(starting_turf, extended_target_turf)
+
+	checked_turfs -= starting_turf
+
+	//At some angles (scatter or otherwise) the original target is not in checked_turfs so we put it in there in order based on distance from user
+	//If we are literally clicking on someone with IFF then we don't want to fire, feels funny as a user otherwise
+	if(projectile_to_fire.original)
+		var/turf/original_target_turf = get_turf(projectile_to_fire.original)
+
+		if(original_target_turf && !(original_target_turf in checked_turfs))
+			var/user_to_target_dist = get_dist(starting_turf, original_target_turf)
+			var/list/temp_checked_turfs = checked_turfs
```

Comment:
This won't work. Doing this is just making a 2nd pointer to the list. Append .Copy() to checked_turfs